project/account/models.py

```diff
@@ -0,0 +1,101 @@
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, username, first_name, last_name, password=None):
+        """
+        Creates and saves a user with the given email, username and password.
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+        if not username:
+            raise ValueError("Users must have a username")
+        if not first_name:
+            raise ValueError("Users must have a first name")
+        if not last_name:
+            raise ValueError("Users must have a last name")
+        if not password:
+            raise ValueError('Users must have a password')
+
+        user = self.model(
+            email=self.normalize_email(email),
+            username=username,
```

Comment:
You listed some fields already as we agreed on, perfect job!
Do you think we should use keyword arguments (**extra_fields) instead so that if we have to add a phone number and address (or other stuff) it would be easier?

Reply:
I just reworked the whole BaseUserManager part to accommodate the **extra_fields